config.go

```diff
@@ -117,9 +119,60 @@ func makeDockerV2S2Image(oimage *ociv1.Image) (docker.Image, error) {
 	return image, nil
 }
 
+func makeDockerV2S1Image(manifest docker.V2S1Manifest) (docker.V2Image, error) {
+	// Treat the most recent (first) item in the history as a description of the image.
+	if len(manifest.History) == 0 {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from manifest")
+	}
+	dimage := docker.V2Image{}
+	err := json.Unmarshal([]byte(manifest.History[0].V1Compatibility), &dimage)
+	if err != nil {
+		return docker.V2Image{}, err
+	}
+	if dimage.DockerVersion == "" {
+		return docker.V2Image{}, fmt.Errorf("error parsing image configuration from history")
+	}
+	// Build a filesystem history.
+	rootFS := &docker.V2S2RootFS{
+		Type:    docker.TypeLayers,
+		DiffIDs: []digest.Digest{},
+	}
+	history := []docker.V2S2History{}
+	for i := range manifest.FSLayers {
+		d := manifest.FSLayers[i].BlobSum
+		// Prepend this layer, because the original manifest's list has the earliest layer last.
+		rootFS.DiffIDs = append([]digest.Digest{d}, rootFS.DiffIDs...)
```

Comment:
It's not an intentional limitation.  I don't see the digest of the uncompressed data in the manifest, though.  If we have to decompress or regenerate the layer data to get the right value, I'd be reluctant to do that.  We compute the right values when we're exporting the image, so if nobody's going to be reading the blatantly-wrong-if-compression-was-used value, then I'd lean hard in the direction of adding a comment about it here and leaving it otherwise.